io/GDALReader.cpp

```diff
@@ -126,81 +131,60 @@ void GDALReader::ready(PointTableRef table)
 }
 
 
-point_count_t GDALReader::read(PointViewPtr view, point_count_t num)
+point_count_t GDALReader::read(PointViewPtr view, point_count_t numPts)
 {
-    point_count_t count = (std::min)(num, m_count - m_index);
-    PointId nextId = view->size();
+    PointId idx = view->size();
+    point_count_t cnt = 0;
+    PointRef point(*view, idx);
+    while (cnt < numPts)
+    {
+        point.setPointId(idx);
+        if (!processOne(point))
+            break;
+        cnt++;
+        idx++;
+    }
+    return cnt;
+}
+
 
-    std::array<double, 2> coords;
-    for (int row = 0; row < m_raster->height(); ++row)
+bool GDALReader::processOne(PointRef& point)
+{
+
+    static std::array<double, 2> coords;
+    if (m_row == m_raster->height() &&
+        m_col == m_raster->width())
+        return false; // done
+
+    if (m_col == m_raster->width())
     {
-        for (int col = 0; col < m_raster->width(); ++col)
-        {
-            m_raster->pixelToCoord(col, row, coords);
-            view->setField(Dimension::Id::X, nextId, coords[0]);
-            view->setField(Dimension::Id::Y, nextId, coords[1]);
-            nextId++;
-        }
+        m_col = 0;
+        m_row ++;
     }
+    m_raster->pixelToCoord(m_col, m_row, coords);
+    double x = coords[0];
+    double y = coords[1];
+    point.setField(Dimension::Id::X, x);
+    point.setField(Dimension::Id::Y, y);
 
-    std::vector<uint8_t> band;
+    static std::vector<double> data;
 
     for (int b = 0; b < m_raster->bandCount(); ++b)
```

Comment:
I don't think you want to do this.  You're calling m_raster->read() for each band, but it returns data for all bands.  You want the read() outside of the band loop.